jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/naming/KeyUtils.java

```diff
@@ -215,7 +210,15 @@ public static String getConcatedTypeNameValues(List<String> typeNames, String ty
 	 */
 	public static String getConcatedTypeNameValues(Query query, List<String> typeNames, String typeName) {
 		Set<String> queryTypeNames = query.getTypeNames();
-		if (queryTypeNames != null && queryTypeNames.size() > 0) {
+		if (query.isUseAllTypeNames()) {
```

Comment:
You something mean like `AllTypeNameValuesStringBuilder`, `QueryFilteredTypeNameValuesStringBuilder` and `DefaultTypeNameValuesStringBuilder` all implementing `TypeNameValuesStringBuilder` interface/abstract class?